CompetionTaskMars/Tests/Hooks.cs

```diff
@@ -0,0 +1,120 @@
+﻿using AventStack.ExtentReports;
+using AventStack.ExtentReports.Reporter;
+using CompetionTaskMars.Helpers;
+using CompetionTaskMars.Pages;
+using CompetionTaskMars.Utilities;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using OpenQA.Selenium;
+using System;
+using System.IO;
+
+namespace CompetionTaskMars.Tests
+{
+    [TestFixture]
+    public class Hooks : Driver
+    {
+        public static ExtentReports extent;
+        public ExtentTest test;
+
+        private SignIn signInObj;
+        private Education educationActions;
+
+        [OneTimeSetUp]
+        public void SetupReport()
+        {
+            var htmlReporter = new ExtentSparkReporter("C:\\repo\\CompetionTaskMars\\Reports\\ExtentReport.html");
+            extent = new ExtentReports();
+            extent.AttachReporter(htmlReporter);
+            htmlReporter.Config.DocumentTitle = "Mars Competition Test Report";
+            htmlReporter.Config.ReportName = "Automation Test Report";
+            htmlReporter.Config.Theme = AventStack.ExtentReports.Reporter.Config.Theme.Standard;
+            extent.AddSystemInfo("UserName", "Kalpana Dissanayake");
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            // Initialize WebDriver and SignIn object
+            driver = InitializeDriver("chrome");
+            test = extent.CreateTest(TestContext.CurrentContext.Test.Name);
+
+            signInObj = new SignIn(driver);
+            educationActions = new Education(driver);
+
+            // Perform Login using credentials from JSON
+            var credentials = Utilities.JsonReader.GetCredentials(); 
+             signInObj.LoginActions(credentials.Email, credentials.Password);            
+            
+        }
+
+        [Test]
+        public void TestAddNewEducation()
+        {
+            var credentials = Utilities.JsonReader.GetCredentials();
+            test.Info("Starting 'Add New Education' test");
+
+            educationActions.NavigateToEducationTab();
+            educationActions.AddNewEducationButtonActions();
+            educationActions.AddCollegeUniversityNameActions(credentials.College);
+            educationActions.AddCountyOfCollegeUniversityActions(credentials.Country);
```

Comment:
this logic either create a step class eg skillsteps which has the method to add education and then you in the test you can just do skillSteps.addEducation or you can just move this to the pages and then in test do educationAction.addEducation